### ISSUE TITLE
Can't find ID in object metadata

### ISSUE THREAD

**Issue by AndyGFX** [Kiamo2/YATI#56]
I have add_id_as_metadata=true and on imported tmx scene in godot ID missing in metadata  obj.get_meta("id") is null. Is it bug or problem with scene setup in Tiled?

**Comment by Kiamo2** [comment]
Hmm, class seems to be added as metadata.
Is the Id of that object perhaps 0?
If yes that could be the problem...
If no I've no clue and would need the .tmx to have a chance to find the problem.

**Comment by AndyGFX** [comment]
Yes class is added but ID not. Strange
I tried fix it and here is hotfix ... maybe not on right place but works.

![image](https://github.com/user-attachments/assets/4d294442-1366-4473-80b9-246c15a0f806)

**Comment by Kiamo2** [comment]
Interesting... ok, by adding it as property it's going to be handled like all properties and thus added to the meta data.
But I wonder if this is the proper way in every respect as an object id is no property but an object attribute.
Therefore, I hesitate to just adopt your hotfix.
I'd rather fundamentally understand why class is added and ID not (class is also an object attribute).
How can I reproduce this behavior?

**Comment by AndyGFX** [comment]
All what you need todo is create any Tiled scene with Objects layer and a few tiles as object, then in Godot on TMX file set "Add ID as Metadata, press Re-import. 
Run scene and via Remote look on tmx node and check generated metadata if ID was created ....

**Comment by Kiamo2** [comment]
In a simple test like you described it everything is  working.
There must be s.th. special with your example, I suspect you're using templates as your fix is part of the template handling.

**Comment by AndyGFX** [comment]
OK. I prepare new project, with simple scene with objects from template in TMX file ... and we see where is problem, because looking for a bug in complex scene is little bit hard ;)

**Comment by AndyGFX** [comment]
YES. Comfirmed, when is tile created from template as object in object layer, then ID is not created ... only CLASS.

**Comment by Kiamo2** [comment]
Yeah, found it.
A template has no id.
But it has a class.

I'll fix that along with a few other inconsistencies I've also found here.
